From 8ab48dab9b174f787b7ccf25139d9b59efb85ba4 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?St=C3=A9phane=20Marchesin?= <marcheu@chromium.org>
Date: Thu, 21 Feb 2013 20:20:37 -0800
Subject: [PATCH 1/4] Disable backlight adjustments on DPMS mode changes.

This prevents the X driver from touching the power_manager-adjuster
backlight levels.
---
 src/intel_display.c | 10 ----------
 1 file changed, 10 deletions(-)

diff --git a/src/intel_display.c b/src/intel_display.c
index 5ee955e..9d7c754 100644
--- a/src/intel_display.c
+++ b/src/intel_display.c
@@ -458,7 +458,6 @@ intel_crtc_apply(xf86CrtcPtr crtc)
 			continue;
 
 		    intel_output = output->driver_private;
-		    intel_output_dpms_backlight(output, intel_output->dpms_mode, DPMSModeOn);
 		    intel_output->dpms_mode = DPMSModeOn;
 		}
 	}
@@ -1051,19 +1050,10 @@ intel_output_dpms(xf86OutputPtr output, int dpms)
 			continue;
 
 		if (!strcmp(props->name, "DPMS")) {
-			/* Make sure to reverse the order between on and off. */
-			if (dpms == DPMSModeOff)
-				intel_output_dpms_backlight(output,
-							    intel_output->dpms_mode,
-							    dpms);
 			drmModeConnectorSetProperty(mode->fd,
 						    intel_output->output_id,
 						    props->prop_id,
 						    dpms);
-			if (dpms != DPMSModeOff)
-				intel_output_dpms_backlight(output,
-							    intel_output->dpms_mode,
-							    dpms);
 			intel_output->dpms_mode = dpms;
 			drmModeFreeProperty(props);
 			return;
-- 
1.8.1.3

